Add support for diffing files

Vivalzar opened the pull request:
I updated this plugin to add support for diffing files (like in ST). It produces that kind of files:
```diff
--- before.py
+++ after.py
@@ -1,4 +1,4 @@
-bacon
-eggs
-ham
+python
+eggy
+hamster
 guido
```

Feel free to try it and to give me some tips: it's my first contribution.

I already know that the new option I added (`diff_context`) doesn't work. It always uses the default value. It would be nice of you if you could help me on that.

veksha commented:
here `ctx.version(0) >= '0.6.8'` is always `False`
(`ctx.version(0) == 0.6.10 2020-10-14`)

```python
def get_opt(key, def_val: tp.Any = ''):
    return ctx.get_opt('differ.' + key, def_val, user_json=JSONFILE) \
           if ctx.version(0) >= '0.6.8' \
           else ctx.get_opt('differ.' + key, def_val)
```

veksha commented:
@Vivalzar , is it correct diff?  (`-ff+ff`)

![image](https://user-images.githubusercontent.com/275333/213343101-52b7db7e-8607-4372-a519-6dd8b9b937b8.png)

Vivalzar commented:
It's because you don't have an empty line at the end of the files. If you add them, the diff output is correct.

I will add them in memory just before diffing.

Vivalzar commented:
Fixed. You can test it again if you want.

veksha commented:
> Fixed. You can test it again if you want.

works now, thanks.

CudaText-addons commented:
@veksha what text must we write in history.txt? i didn't try changes.

Vivalzar commented:
> I already know that the new option I added (`diff_context`) doesn't work. It always uses the default value. It would be nice of you if you could help me on that.

To my knowledge, the new option I added `diff_context` still doesn't work...
There is a problem with `ctx.version(0)` apparently...

Vivalzar commented:
@Alexey-T I don't know if you're notified here, so I'm pinging you directly.

How to correct the bug with the version? Should someone change the version `0.6.8`? Should I open an issue on the main repo?

Alexey-T commented:
@Vivalzar i am lost in this topic. pls formulate again: what is a problem with version? do you see it after my last commits?

Vivalzar commented:
@Alexey-T Ok, I'm very sorry, it actually works correctly. I didn't see your commit. Forget about this. Again, I'm sorry.
